### PR TITLE
[STRATCONN-1146] Expose Flagon and DataDog stats client

### PR DESCRIPTION
**Pull request by brennan**
- Jira: https://segment.atlassian.net/browse/STRATCONN-1146
- This PR updates actions `destination-kit` to accept an instance of the Stats client and Flagon features, both passed from the monoservice in this PR: https://github.com/segmentio/integrations/pull/2297
- Testing completed successfully locally: https://paper.dropbox.com/doc/Test-Doc-Flagon-and-DataDog-in-Action-Destinations--BkDeVPbZOGa7NX4DJV2WZ99MAg-r57m6xS4UpBZkBnCwx8FT

## Testing

_Include any additional information about the testing you have completed to
ensure your changes behave as expected. For a speedy review, please check
any of the tasks you completed below during your testing._

- [x] Added [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
- [x] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)
- [x] [Segmenters] Tested in the staging environment

### CODE REVIEW

File: packages/core/src/destination-kit/types.ts [hunk 1]
```diff
@@ -23,6 +23,8 @@ export interface ExecuteInput<Settings, Payload> {
   page?: string
   /** The data needed in OAuth requests */
   readonly auth?: AuthTokens
+  /** The Flagon features available in the request */
```
Comment (mericsson) [inline packages/core/src/destination-kit/types.ts:26]:
```suggestion
  /** The features available in the request based on customer workspace */
```

I think including word Flagon may confuse people.

**Comment by mericsson** [comment]
As part of PR can you add a note to the appropriate place in README about these and how / if internal vs external developer can access.

**Review by rhall-twilio — APPROVED** [review]

**Review by marinhero — APPROVED** [review]

File: packages/core/src/destination-kit/action.ts [hunk 2]
```diff
@@ -76,6 +76,7 @@ interface ExecuteBundle<T = unknown, Data = unknown> {
   settings: T
   mapping: JSONObject
   auth: AuthTokens | undefined
+  features?: { [key: string]: boolean }
```
Comment (dlasky) [inline packages/core/src/destination-kit/action.ts:79]: consider adding a JSDoc comment here that this is for internal use only so it shows up in vscode. We could also consider misusing deprecation here as a warning as well.

File: packages/core/src/destination-kit/index.ts [hunk 3]
```diff
@@ -160,6 +160,7 @@ interface EventInput<Settings> {
   readonly settings: Settings
   /** Authentication-related data based on the destination's authentication.fields definition and authentication scheme */
   readonly auth?: AuthTokens
+  readonly features?: { [key: string]: boolean }
```
Comment (dlasky) [inline packages/core/src/destination-kit/index.ts:163]: JSDoc as above

**Comment by brennan** [comment]
@dlasky Will do - thanks, Dan!

**Pull request merged**